backend/open_webui/main.py

```
@@ -1160,9 +1328,16 @@ async def healthcheck():
-    Session.execute(text("SELECT 1;")).all()
-    return {"status": True}
+def healthcheck_with_db():
+    try:
+        Session.execute(text("SELECT 1;")).all()
+        return {"status": True}
+    except Exception as e:
+        log.error(f"Database health check failed: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+            detail="Database connection failed",
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)